### PR TITLE
Fix_Readd_Improved_Combat | Readded improved combat scripts

### PR DESCRIPTION
**Pull request by RyyzQ**
Readded improved Combat scripts.

### CODE REVIEW

File: Assets/_Scenes/Scene_Combat/Scripts/SequenceHandler.cs [hunk 1]
```diff
@@ -32,7 +31,8 @@ public SequenceState GetCurrentSequenceState()
     public void MoveCurrentSequenceToBottom()
```
Comment (Turbotondel) [inline Assets/_Scenes/Scene_Combat/Scripts/SequenceHandler.cs]: Explanation of method is unclear in regard of its function, maybe consider rewriting this.
I get what it does, but the method name and description both describe it in a different way, as the name refers to the current sequence and the description refers to the next one

File: Assets/_Scenes/Scene_Combat/Scripts/NodeController.cs [hunk 2]
```diff
@@ -98,149 +97,120 @@ private void Start()
...
+                    //(during the same update cycle)
+                    CancelInvoke("CreateNote");
+                    beatTimer = CalculateBeatTimer(CurrentSequence.GetCurrentSpawnedPattern().PatternLength);
+                    InvokeRepeating("CreateNote", 0f, beatTimer);
+                } 
             }
         }
 
-        if (CurrentSequence.SequenceState == SequenceState.Attack && CurrentSequence.IsEndOfActivePattern)
+        if (CurrentSequence.IsSequenceDone(CurrentSequence.SequenceState, beatCounter))
```
Comment (mrtsven) [inline Assets/_Scenes/Scene_Combat/Scripts/NodeController.cs]: Add some comments, it's a bit unclear what's happening

File: Assets/_Scenes/Scene_Combat/Scripts/NodeController.cs [hunk 3]
```diff
@@ -98,149 +97,120 @@ private void Start()
...
+            SetAudioClip(CurrentSequence.patterns[CurrentSequence.SpawnedPatternIndex].AudioFileName);
+            CancelInvoke("CreateNote");
+            beatTimer = CalculateBeatTimer(CurrentSequence.GetCurrentSpawnedPattern().PatternLength);
             PatternScore = 0;
+            InvokeRepeating("CreateNote", 0f, beatTimer);
         }
 
-        if (CurrentSequence.IsSequenceDone(CurrentSequence.SequenceState, beatCounter))
+
+        if (audioSource.time >= (audioSource.clip.length - 0.05f))
```
Comment (mrtsven) [inline Assets/_Scenes/Scene_Combat/Scripts/NodeController.cs]: Idem, some comments could clarify what's happening

File: Assets/_Scenes/Scene_Combat/Scripts/NoteBehaviour.cs [hunk 4]
```diff
@@ -18,14 +19,21 @@ private void OnEnable()
         Invoke("Destroy", destroyTime);
     }
 
+    public void SetParentPattern(Pattern pattern)
```
Comment (mrtsven) [inline Assets/_Scenes/Scene_Combat/Scripts/NoteBehaviour.cs]: Clarify this through a comment, it's a unclear what this does.

**Pull request merged**